mr/src/main/java/org/elasticsearch/hadoop/rest/query/RawQueryBuilder.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.hadoop.rest.query;
+
+import org.codehaus.jackson.JsonParser;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.elasticsearch.hadoop.EsHadoopIllegalArgumentException;
+import org.elasticsearch.hadoop.serialization.Generator;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class RawQueryBuilder extends QueryBuilder {
+    private static final ObjectMapper MAPPER =
+            new ObjectMapper()
+                    .configure(JsonParser.Feature.ALLOW_COMMENTS, true);
+    private final String queryString;
+
+    public RawQueryBuilder(String value) throws IOException {
+        this((Map<String, Object>) MAPPER.readValue(value, HashMap.class));
+    }
+
+    public RawQueryBuilder(Map<String, Object> map) throws IOException {
+        Object query = map;
+        if (map.containsKey("query")) {
+            query = map.remove("query");
```

Comment:
This line right here breaks backwards compatibility with ES v1.7.3 for SparkSQL. We have code that hooks into Spark that allows us to make sense of some of the SQL operations being executed and translate those operations into pushdown filters and queries. 

In the case of running against ES 1.7.3, the framework will detect the version and will create a query context with the pushdown query underneath it to allow compatibility with the filter context. Because all the filters are passed through the parsing code listed above before building the final query, the query context marker is removed from the JSON and the query ceases to compile from within the filter context.

Basically, the filter fragment goes from being this:
```
{ "query": { "match" : { "..." : "..." } } }
```
to this:
```
{ "match": { "..." : "..." } }
```
which is then inserted into a filter context and subsequently comes back from the server with the following error: 
```
org.elasticsearch.search.SearchParseException: [defaultstrictspark-test][2]: from[-1],size[-1]: Parse Failure [Failed to parse source [{"query":{"filtered":{"query":{"match_all":{}},"filter":{"match":{"airport":"OTP"}}}}}]]
	at org.elasticsearch.search.SearchService.parseSource(SearchService.java:747)
	<...>
Caused by: org.elasticsearch.index.query.QueryParsingException: [defaultstrictspark-test] No filter registered for [match]
```

This probably isn't a huge blocker for the alpha release, but it is a big deal if we want to continue the support for 1.x.